pkg/internal/itest/bitswapfetch_test.go

```diff
@@ -0,0 +1,172 @@
+package itest
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/index-provider/metadata"
+	"github.com/filecoin-project/lassie/pkg/internal/itest/testpeer"
+	"github.com/filecoin-project/lassie/pkg/internal/testutil"
+	"github.com/filecoin-project/lassie/pkg/lassie"
+	httpserver "github.com/filecoin-project/lassie/pkg/server/http"
+	"github.com/filecoin-project/lassie/pkg/types"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-graphsync/storeutil"
+	"github.com/ipfs/go-libipfs/bitswap/network"
+	"github.com/ipfs/go-libipfs/bitswap/server"
+	"github.com/ipfs/go-unixfsnode"
+	"github.com/ipfs/go-unixfsnode/data/builder"
+	"github.com/ipld/go-car/v2/storage"
+	dagpb "github.com/ipld/go-codec-dagpb"
+	"github.com/ipld/go-ipld-prime/linking"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/libp2p/go-libp2p/core/peer"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBitswapFetchTwoPeers(t *testing.T) {
+	req := require.New(t)
+	mn := mocknet.New()
+	ctx := context.Background()
+	testPeerGenerator := testpeer.NewTestPeerGenerator(ctx, mn, []network.NetOpt{}, []server.Option{})
+	peers := testPeerGenerator.Peers(2)
+
+	ls := storeutil.LinkSystemForBlockstore(peers[0].Blockstore())
```

Comment:
This wizardry deserves comments for mortals.

```suggestion
	// build two files of 4MiB random bytes, packaged into unixfs DAGs (root1 & root2)
	// and the original source data retained (srcData1, srcData2)
	ls := storeutil.LinkSystemForBlockstore(peers[0].Blockstore())
```